server/views/pages/recommendations/checkBookingDetails.njk

```diff
@@ -302,16 +301,17 @@
                             Last known address
                             <p class="govuk-hint">From the Part A</p>
                         </div>
-                        {% if recommendation.personOnProbation.addresses | length == 0 and not recommendation.isMainAddressWherePersonCanBeFound.details %}
+                        {% if hasLastKnownAddress and not recommendation.isMainAddressWherePersonCanBeFound.details %}
                             <div class="govuk-summary-list__value_middle"><br />- <span class="govuk-visually-hidden">no value</span></div>
                         {% else %}
                             <div class="govuk-summary-list__value">
+                                {% if hasLastKnownAddress %}
```

Comment:
Looks like something might not be right here.  It's saying if there is a "last known address", display "-" AND display the addresses in recommendation.personOnProbation.addresses.

I think the logic of hasLastKnownAddress is inverted.

Reply:
Have refactored and added a utility function and unit tests